### PR TITLE
Deploy in docker container

### PR DESCRIPTION
**Pull request by maximsolodukhin**

### CODE REVIEW

File: routers/algorithm.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#     df[['id', 'name', 'lat', 'long']] = df.apply(lambda row: pd.Series({'id': row['id'], 'name': row['name'], 'lat': row['lat'], 'long': row['long']}), axis=1)
+#     return df
+
+# def get_coordinates(df):
+#     df['long'], df['lat'] = zip(*df['location'].apply(lambda loc: (loc.longitude, loc.latitude)))
+#     coordinates = df[['long', 'lat']].values
+#     return coordinates
+
+def get_time_matrix(coordinates):
+    api = rp.Graphhopper(api_key=graphhopper_api_key)
```
Comment (nikita-kostin) [inline routers/algorithm.py]: Client construction might be an expensive operation, I would say we should move this code outside the function to initialise it just once

File: routers/algorithm.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    m_symm = np.concatenate((m_symm_top, m_symm_bottom), axis=0)
+    return m_symm.astype(int) # Concorde requires integer weights
+
+
+def solve_tcp(symmetric_matrix, comeback: bool = False):
+    if not comeback:
+        symmetric_matrix[:, 0] = 0 #doesn't come back to the start
+    permutation, distance = solve_tsp_dynamic_programming(symmetric_matrix)
+    tour = permutation[::2]
+    return tour, distance
```
Comment (nikita-kostin) [inline routers/algorithm.py]: `distance` return value is unused - at least so far - we can remove it

File: routers/algorithm.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import routingpy as rp
+from python_tsp.exact import solve_tsp_dynamic_programming
+import os
+from fastapi import APIRouter, Depends
+from models import supabase, User
+from shapely import wkb
+
+router = APIRouter(prefix="/algorithm", tags=["Algorithm"])
+
+graphhopper_api_key = os.environ.get("GRAPHOPPER_API_KEY", None)
```
Comment (nikita-kostin) [inline routers/algorithm.py:14]: We need to pass this in deployment scripts and set the values in the GitHub secrets. Might also be done as a follow-up PR, no strong preference

**Review by nikita-kostin — CHANGES_REQUESTED** [review]

File: routers/algorithm.py (hunk 3 above)
Comment (artemmelnychuk) [reply routers/algorithm.py:14]: Please set the GRAPHOPPER_API_KEY value in the Github

**Review by nikita-kostin — APPROVED** [review]

**Pull request merged**